tasks/gcp/common.py

```diff
@@ -0,0 +1,57 @@
+from typing import List, Union
+
```

Comment:
Can probably be done in a follow up PRs. But I feel like most of these functions could defined in a `tasks/common.py` file so we do not need to have it duplicated for each cloud provider

Reply:
Some of them only in fact because the default architecture, os, etc can be different. But some of them could, I'll have a look!